client/shared/src/util/url.ts

```diff
@@ -654,5 +685,27 @@ export const isExternalLink = (url: string): boolean =>
 export const appendSubtreeQueryParameter = (url: string): string => {
     const newUrl = new URL(url, window.location.href)
     newUrl.searchParams.set('subtree', 'true')
-    return newUrl.pathname + newUrl.search + newUrl.hash
+    return newUrl.pathname + `?${formatSearchParameters(newUrl.searchParams)}` + newUrl.hash
+}
+
+/**
+ * Stringifies the provided search parameters, replaces encoded `/` and `:` characters,
+ * and removes trailing `=`.
+ */
+export const formatSearchParameters = (searchParameters: URLSearchParams): string =>
+    searchParameters.toString().replace(/%2F/g, '/').replace(/%3A/g, ':').replace(/=&/g, '&').replace(/=$/, '')
+
+export const addLineRangeQueryParameter = (searchParameters: URLSearchParams, range: string): URLSearchParams => {
```

Comment:
If `range` can be empty (as indicated by the check below) it would be better/more idiomatic to mark it as `string | undefined` so that the check is enforced and pass `undefined` from the caller instead of an empty string. Empty strings are generally avoided in TS because they can't be type-checked, if a parameter is called and typed `range: string` one would usually expect that it is always a non-empty string of a valid range.
```suggestion
export const addLineRangeQueryParameter = (searchParameters: URLSearchParams, range: string | undefined): URLSearchParams => {
```

Reply:
Good point, the problem was in the `toPositionOrRangeQueryParameter` where we returned an empty string if nothing matched.

Reply:
Heads up, the result of `toPositionOrRangeQueryParameter` was inlined into a string in one place, which got caught by our listing in CI: https://buildkite.com/sourcegraph/sourcegraph/builds/99323#db933df2-cb44-487a-a928-308b8fa5946b/125-128
You just need to add a `?? ''` wherever `undefined` is potentially added to a string.

Reply:
Thank god for CI 😄 Should be fixed now.

Reply:
All hail our lord and savior, ESLint